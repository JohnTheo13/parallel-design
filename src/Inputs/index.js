export { Input, Checkbox, Radio } from './components';
