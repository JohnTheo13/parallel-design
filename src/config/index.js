import themeProps from './utils/themeProps';
import whitespace from './utils/spacing';
import * as colors from './utils/colors';
import theme from './theme';

import {
	borderRadius,
	borderTopRadius,
	borderRightRadius,
	borderBottomRadius,
	borderLeftRadius
} from './utils/border-radius';

import {
	hover,
	hoverFocus,
	plainHoverFocus,
	hoverFocusActive
} from './utils/hover';

import { sassRgba } from './utils/sassRgba';
import { boxShadow } from './utils/box-shadow';
import { transition } from './utils/transition';
import { gradientStriped } from './utils/gradients';
import * as fonts from './utils/fonts';

export {
	themeProps,
	theme,
	sassRgba,
	boxShadow,
	transition,
	borderRadius,
	borderTopRadius,
	borderRightRadius,
	borderBottomRadius,
	borderLeftRadius,
	gradientStriped,
	hover,
	hoverFocus,
	plainHoverFocus,
	hoverFocusActive,
	whitespace,
	colors,
	fonts
};
