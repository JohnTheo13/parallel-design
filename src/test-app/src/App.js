import React from 'react';
import { Button } from 'buttons';
import CheckTest from './Components/CheckboxCheck';
import './App.css';

function App() {
	return (
		<div className="App">
			<CheckTest />
			<Button tag="primary" height="68px" shape="sharp">
				Primary
			</Button>
		</div>
	);
}

export default App;
