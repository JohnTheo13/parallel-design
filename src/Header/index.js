// @flow

import { createElement, type Node } from 'react';
import styled from 'styled-components';
import { theme as defaultTheme, themeProps } from '../config';

const { header } = defaultTheme;

type Props = {
	as: string,
	children: Node,
	color?: string
};

const Element = ({ as, children, ...props }: Props) =>
	createElement(as, props, children);

const Header = styled(Element)`
	margin-bottom: ${({ as }) =>
		themeProps(`header.${as}.marginButton`, header[as].marginBottom)};
	font-size: ${({ as }) =>
		themeProps(`header.fontSize.${as}`, header[as].fontSize)};
	font-family: ${({ as }) =>
		themeProps(`header.${as}.fontFamily`, header[as].fontFamily)};
	font-weight: ${({ as }) =>
		themeProps(`header.${as}.fontWeight`, header[as].fontWeight)};
	line-height: ${({ as }) =>
		themeProps(`header.${as}.lineHeight`, header[as].lineHeight)};
	color: ${({ color, as }) =>
		color || themeProps(`header.${as}.color`, header[as].color)};
`;

export default Header;
