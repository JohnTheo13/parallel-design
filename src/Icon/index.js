// @flow
import React from 'react';
import styled from 'styled-components';
import { colors as col } from '../config';
import icons from './Icons';

type IconType = {
	viewBox?: string,
	style?: Object,
	color?: {
		default: string,
		hover: string
	},
	hover?: boolean,
	width?: string,
	height?: string,
	onClick?: Function,
	onMouseOver?: Function,
	onMouseOut?: Function,
	className?: string,
	name: string
};

const StyledSvg = styled.svg`
	fill: ${({ color, hover }) => (hover ? color.hover : color.default)};
	height: ${({ height }) => `${height}px`};
	width: ${({ width }) => `${width}px`};
	&:hover {
		fill: ${({ color }) => color.hover};
	}
`;

const Icon = ({
	viewBox,
	style,
	color,
	className,
	onClick,
	onMouseOver,
	onMouseOut,
	width,
	height,
	name,
	hover
}: IconType) => (
	<StyledSvg
		version="1.1"
		x="0px"
		y="0px"
		viewBox={viewBox}
		style={style}
		color={color}
		className={className}
		onClick={onClick}
		onMouseOver={onMouseOver}
		onFocus={onMouseOver}
		onMouseOut={onMouseOut}
		height={height}
		width={width}
		onBlur={onMouseOut}
		id={name}
		hover={hover}
	>
		{icons(name)}
	</StyledSvg>
);

Icon.defaultProps = {
	viewBox: '0 0 48 48',
	style: undefined,
	color: {
		default: col.dark,
		hover: col.dark
	},
	hover: false,
	className: '',
	onClick: undefined,
	onMouseOver: undefined,
	onMouseOut: undefined,
	width: '20',
	height: '20'
};

export default Icon;
