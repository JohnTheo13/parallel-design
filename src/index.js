import { Button } from './Buttons';
import Header from './Header';
import Icon from './Icon';
import { Input, Checkbox, Radio } from './Inputs';

export { Button, Header, Icon, Input, Checkbox, Radio };
