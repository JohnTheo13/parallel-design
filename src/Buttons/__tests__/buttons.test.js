import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Button } from '../../src';

test('it has hover color', () => {
	const button = renderer
		.create(
			<Button
				tag="primary"
				height="80px"
				width="150px"
				shape="rounded"
				disabled={false}
			>
				Interactive
			</Button>
		)
		.toJSON();
	expect(button).toHaveStyleRule('background-color', '#40a9ff', {
		modifier: ':hover'
	});
});

test('Primary button snap', () => {
	const button = renderer
		.create(
			<Button
				tag="primary"
				height="80px"
				width="150px"
				shape="rounded"
				disabled={false}
			>
				Interactive
			</Button>
		)
		.toJSON();
	expect(button).toMatchSnapshot();
});
