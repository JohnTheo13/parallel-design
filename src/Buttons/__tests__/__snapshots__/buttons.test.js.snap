// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button snap 1`] = `
.c0 {
  font-family: initial;
  font-size: 1rem;
  font-weight: normal;
  color: #fff;
  background-color: #1890ff;
  border: 1px solid #1890ff;
  border-radius: 2px;
  cursor: pointer;
  height: 80px;
  width: 150px;
}

.c0:disabled {
  color: #fff;
  background-color: #ced4da;
  border-color: #ced4da;
}

.c0:hover {
  outline: none;
  background-color: #40a9ff;
}

<button
  className="c0"
  disabled={false}
  height="80px"
  shape="rounded"
  type="button"
  width="150px"
>
  Interactive
</button>
`;

exports[`it works 1`] = `
.c0 {
  font-family: initial;
  font-size: 1rem;
  font-weight: normal;
  color: #fff;
  background-color: #1890ff;
  border: 1px solid #1890ff;
  border-radius: 2px;
  cursor: pointer;
  height: 80px;
  width: 150px;
}

.c0:disabled {
  color: #fff;
  background-color: #ced4da;
  border-color: #ced4da;
}

.c0:hover {
  outline: none;
  background-color: #40a9ff;
}

<button
  className="c0"
  disabled={false}
  height="80px"
  shape="rounded"
  type="button"
  width="150px"
>
  Interactive
</button>
`;
